pkg/internal/discover/finder.go

```diff
@@ -83,27 +82,16 @@ func newCommonTracersGroup(cfg *beyla.Config) []ebpf.Tracer {
 		return []ebpf.Tracer{httptracer.New(cfg)}
 	}
 
-	tracers := []ebpf.Tracer{}
-
 	switch cfg.EBPF.ContextPropagation {
 	case config.ContextPropagationAll:
-		if tctracer.CanRun() {
-			tracers = append(tracers, tctracer.New(cfg))
-		} else {
-			slog.Warn("L4 trace context propagation cannot be enabled, missing host PID or host network access")
-		}
-		tracers = append(tracers, tpinjector.New(cfg))
+		return []ebpf.Tracer{tctracer.New(cfg), tpinjector.New(cfg)}
```

Comment:
I mentioned this on the original PR, we can't add the tptracer here if we can't run it. later we'll attempt to load it and crash.